### PR TITLE
devenv: remove legacy ensure-venv

### PR DESCRIPTION
**Pull request by joshuarli**
this removes a bunch of legacy code surrounding ensure-venv stuff by requiring devenv (it's already required in .envrc) and replacing it with a much simpler VIRTUAL_ENV check

also good with bootstrap here: https://github.com/getsentry/devenv/pull/109

### CODE REVIEW

File: scripts/ensure-venv.sh
Comment (joshuarli) [inline scripts/ensure-venv.sh:1]: `scripts/bootstrap-py3-venv` was removed a long time ago!

File: scripts/do.sh [hunk 1]
```diff
@@ -2,15 +2,28 @@
...
 # Call this script as "do.sh method_from_lib" to execute any function from that library
 set -eu
+
+command -v devenv >/dev/null 2>&1 || {
+    echo '
+Please install the devenv tool:
+https://github.com/getsentry/devenv#install
+'
+    exit 1
+}
```
Comment (asottile-sentry) [inline scripts/do.sh:12]: these should be `if` statements rather than `||` pretending to be an `if`

File: scripts/do.sh [hunk 2]
```diff
@@ -2,15 +2,28 @@
...
+
 HERE="$(
     cd "$(dirname "${BASH_SOURCE[0]}")"
     pwd -P
 )"
```
Comment (asottile-sentry) [inline scripts/do.sh:17]: ~usually this is first in the file shrugs

File: scripts/do.sh (hunk 1 above)
Comment (joshuarli) [reply scripts/do.sh:12]: yep, bad habit heh

File: scripts/lib.sh [hunk 3]
```diff
@@ -28,48 +26,10 @@ export XDG_CONFIG_DIRS="${XDG_CONFIG_DIRS:-/etc/xdg}"
 export XDG_CACHE_HOME="${XDG_CACHE_HOME:-$HOME/.cache}"
 export XDG_RUNTIME_DIR="${XDG_RUNTIME_DIR:-/var/run}"
 
-
-# Check if a command is available
 require() {
     command -v "$1" >/dev/null 2>&1
 }
 
-query-valid-python-version() {
```
Comment (asottile-sentry) [inline scripts/lib.sh:37]: this might still be useful -- I think there's still ways people can manually get out of sync ?

**Review by asottile-sentry — APPROVED** [review]

File: scripts/do.sh [hunk 4]
```diff
@@ -16,11 +16,20 @@ https://github.com/getsentry/devenv#install
     exit 1
 fi
 
-if ! [[ "$VIRTUAL_ENV" -ef "${HERE}/../.venv" ]]; then
-    echo "
-Your virtualenv isn't activated. You need to successfully run 'direnv allow'.
+if [[ ${GETSENTRY:-0} -eq 1 ]]; then
```
Comment (joshuarli) [inline scripts/do.sh:19]: see https://github.com/getsentry/getsentry/pull/13540

now that i'm revisiting all of this it's pretty annoying how much getsentry make has drifted...

really need to think about how we can move as much of this stuff over to devenv

**Review by asottile-sentry — APPROVED** [review]

**Review by anonrig — APPROVED** [review]

File: scripts/do.sh [hunk 5]
```diff
@@ -16,20 +16,11 @@ https://github.com/getsentry/devenv#install
     exit 1
 fi
 
-if [[ ${GETSENTRY:-0} -eq 1 ]]; then
```
Comment (joshuarli) [inline scripts/do.sh]: soooo this is still needed, but not if i move make setup-git into `devenv sync`: https://github.com/getsentry/getsentry/pull/13540/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R28

**Pull request closed without merging**